File: core/clientLaya/game/src/commonGame/tool/func/PlayerRankTool.ts
namespace Shine
{
    /** 数据注册类 */
    export class PlayerRankTool extends PlayerFuncTool
    {
            /** 数据 */
        private _rankData:PlayerRankData;

        /** 翻页显示插件 */
        private _pageShowTool:PlayerPageShowTool;

        constructor(funcID:number)
        {
            super(FuncToolType.Rank,funcID)
        }

        /** 获取显示排行 */
        public getRank():number
        {
            return this._rankData.rank;
        }

        /** 获取排行数据 */
        public getRankData():PlayerRankData 
        {
            return this._rankData;
        }

        protected toSetData(data:FuncToolData ):void
        {
            super.toSetData(data);

            var dd:RankSimpleData=(data as RankSimpleData);

            this._rankData=GameC.factory.createPlayerRankData();
            this._rankData.key=this.me.role.getPlayerID();
            this._rankData.showData=this.me.role.getSelfRoleShowData();

            this._rankData.rank=dd.rank;
           this. _rankData.value=dd.value;
        }

        /** 刷新排行 */
        public onRefreshRank(rank:number,value:number):void
        {
            this._rankData.rank=rank;
            this._rankData.value=value;

            this.me.dispatch(GameEventType.FuncRefreshRank,this._funcID);
        }

        /** 重置排行榜 */
        public onResetRank():void
        {
            this._rankData.value=0;
            this._rankData.rank=-1;
            
            this.me.dispatch(GameEventType.FuncRefreshRank,this._funcID);
        }

        /** 绑定翻页显示插件 */
        public bindPageShowTool(eachShowNum:number):void
        {
           this.me.func.registFuncTool(this._pageShowTool=new PlayerPageShowTool(this._funcID,eachShowNum));
        }

        /** 获取翻页显示插件 */
        public getPageShowTool():PlayerPageShowTool 
        {
            return this._pageShowTool;
        }
    }
}

File: core/client/game/src/commonGame/tool/func/PlayerRankTool.cs
using System;
using ShineEngine;

/// <summary>
/// 角色排行插件
/// </summary>
public class PlayerRankTool:PlayerFuncTool
{
	/** 数据 */
	private PlayerRankData _rankData;

	/** 翻页显示插件 */
	private PlayerPageShowTool _pageShowTool;

	public PlayerRankTool(int funcID):base(FuncToolType.Rank,funcID)
	{
		
	}

	/** 获取显示排行 */
	public int getRank()
	{
		return _rankData.rank;
	}

	/** 获取排行数据 */
	public PlayerRankData getRankData()
	{
		return _rankData;
	}

	protected override void toSetData(FuncToolData data)
	{
		base.toSetData(data);

		RankSimpleData dd=(RankSimpleData)data;

		_rankData=GameC.factory.createPlayerRankData();
		_rankData.key=me.role.playerID;
		_rankData.showData=me.role.getSelfRoleShowData();

		_rankData.rank=dd.rank;
		_rankData.value=dd.value;
	}

	/** 刷新排行 */
	public void onRefreshRank(int rank,long value)
	{
		_rankData.rank=rank;
		_rankData.value=value;

		me.dispatch(GameEventType.FuncRefreshRank,_funcID);
	}

	/** 重置排行榜 */
	public void onResetRank()
	{
		_rankData.value=0L;
		_rankData.rank=-1;
		
		if(_pageShowTool!=null)
			_pageShowTool.clear();
		
		me.dispatch(GameEventType.FuncResetRank,_funcID);
	}

	/** 绑定翻页显示插件 */
	public void bindPageShowTool(int eachShowNum)
	{
		me.func.registFuncTool(_pageShowTool=new PlayerPageShowTool(_funcID,eachShowNum));
	}

	/** 获取翻页显示插件 */
	public PlayerPageShowTool getPageShowTool()
	{
		return _pageShowTool;
	}
}